cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1470,12 +1470,16 @@
 (rule (lower (icmp (IntCC.SignedLessThan) a @ (value_type (vec128 ty)) b))
       (x64_pcmpgt ty b a))
 (rule (lower (icmp (IntCC.UnsignedGreaterThan) a @ (value_type (vec128 ty)) b))
-      (let ((max Xmm (x64_pmaxu ty a b))
+      ;; N.B.: we must manually prevent load coalescing of these operands; the
+      ;; register allocator gets confused otherwise. TODO:
+      ;; https://github.com/bytecodealliance/wasmtime/issues/3953.
+      (let ((max Xmm (x64_pmaxu ty (put_in_xmm a) (put_in_xmm b)))
```

Comment:
I would probably write this as `(a_reg Xmm (put_in_xmm a))` and then use `a_reg` afterward. The use of `b` below feels a bit asymmetrical and confusing otherwise.

Reply:
Yeah, good catch; I was doing this half-implicit/half-explicit earlier (and thanks to @alexcrichton also for pointing that out). Fixed in c8d8556.